Add clear all unselected button to compare menu

JAssertz opened the pull request:
Changelog: Added "Clear all unselected" button to compare menu.

Resolves #11484 

Before click
<img width="1909" height="426" alt="image" src="https://github.com/user-attachments/assets/ee0fde5f-0b96-47ab-87ea-96762c52b33a" />

After click
<img width="1913" height="432" alt="image" src="https://github.com/user-attachments/assets/50de447e-dc68-404c-81ec-6ffbed2e5889" />

bhollis commented on src/app/compare/Compare.tsx:
```diff
@@ -304,12 +304,27 @@ export default function Compare({ session }: { session: CompareSession }) {
+        <button
+          type="button"
+          className="dim-button"
+          onClick={() => {
+            for (const item of compareItems) {
+              if (item !== initialItem) {
+                remove(item);
+              }
+            }
+          }}
```
I generally prefer to pull click handlers out into named functions instead of having them inline.

bhollis reviewed (commented):
Code looks fine, the button itself feels too large with all those words. I was thinking it could be an icon, though it shouldn't be confused with the close sheet icon.

chainrez commented:
I am not really into the button but whatever.

It is really wordy and ambiguous though. If implemented, it should probably just be a leftmost button that says "This Item", maybe with the item's icon.

JAssertz commented:
> I am not really into the button but whatever.
> 
> It is really wordy and ambiguous though. If implemented, it should probably just be a leftmost button that says "This Item", maybe with the item's icon.

I actually think I prefer this over the original implementation since its more flexible, but I'm curious what Ben thinks.

bhollis commented:
Yeah I think that's right

JAssertz commented:
<img width="1908" height="423" alt="image" src="https://github.com/user-attachments/assets/7de91bf7-813d-431d-93d8-af433a8f53df" />

bhollis commented on src/app/compare/compare-buttons.tsx:
```diff
@@ -279,6 +286,12 @@ export function findSimilarWeapons(exampleItem: DimItem): CompareButton[] {
     },
+    // Exact weapon based on ID
+    exampleItem.id !== '0' && {
+      // Remove this option for D1 items
```
This comment doesn't seem to belong?

JAssertz commented:
> At best it would be "Initial Item" which is still too wordy. I'd love to get this down to just an icon, which people can learn the effect of after pressing it once or twice. Maybe something like https://fontawesome.com/v4/icon/step-backward.
> 
> Using the item ID filter is totally fine.

When you mention the Icon, are you thinking about keeping it as it currently is, but replacing `This item` with the icon, or adding an icon before the `X` to close the sheet using that icon?

chainrez commented on src/app/compare/compare-buttons.tsx:
```diff
@@ -286,6 +286,12 @@ export function findSimilarWeapons(exampleItem: DimItem): CompareButton[] {
       query: compareNameQuery(exampleItem),
     },
+    // Exact weapon based on ID
+    exampleItem.id !== '0' && {
+      // Remove this option for D1 items
```
A comparable item is never going to have id 0. You may have misunderstood the DimItem.id docs?
A non-instanced item is like a stack of coins. A weapon or armor is instanced, there's a specific copy with power, perks, etc. I'm not sure why you would want to exclude this button in D1 anyway.

JAssertz commented:
Is this what you are envisioning, or were you thinking about a button that looks like and sits next to the close sheet button? If this was the goal do we want to remove the (1) since it will always be 1?

<img width="1385" height="386" alt="image" src="https://github.com/user-attachments/assets/74e0bbda-04ba-4c27-9698-15bd6948bdd3" />

JAssertz commented:
<img width="1390" height="273" alt="image" src="https://github.com/user-attachments/assets/ad5828fe-447a-45c9-b259-782435185b20" />

added compatibility for armor and ships/sparrows/ghost shells, and removed the item counter text.

bhollis commented:
It's a tough thing to communicate in an icon, I was hoping that would be "reset to left only" or something. An eraser could work too - I think the broom might confuse people who expect some "vault clean" feature.

bhollis commented on src/app/compare/CompareSuggestions.tsx:
```diff
@@ -83,7 +87,7 @@ export default memo(function CompareSuggestions({
         >
           {buttonLabel.map((l) => (typeof l === 'string' ? <span key={l}>{l}</span> : l))}
-          <span key="itemcount">({items.length})</span>
+          {!query.includes('id:') && <span>({items.length})</span>}
```
I think the query will include id if you manually exclude any item by clicking the X on it
